tsdownsample/downsampling_interface.py

```diff
@@ -152,9 +200,13 @@ def _switch_mod_with_x_and_y(
     raise ValueError(f"Unsupported data type (for x): {x_dtype}")
 
 
-class RustDownsamplingInterface(DownsampleInterface):
-    def __init__(self, name: str, resampling_mod: ModuleType) -> None:
-        super().__init__(name + " [tsdownsample_rs]")
+class AbstractRustDownsampler(AbstractDownsampler, ABC):
```

Comment:
Why are the `__switch_mod_with_y` and `_switch_mod_with_x_and_y` embedded in this class?

Reply:
@jvdd, what is your opinion?

Reply:
Agree, will make them static methods